src/rdydm.c

```diff
@@ -0,0 +1,221 @@
+#include <petscdmceed.h>
+#include <petscdmplex.h>
+#include <private/rdycoreimpl.h>
+#include <rdycore.h>
+
+// Maximum length of the name of a prognostic or diagnostic field component
+#define MAX_COMP_NAME_LENGTH 20
+
+PetscErrorCode CloneAndCreateCellCenteredDM(DM dm, PetscBool is_dm_refined, PetscInt n_aux_field, PetscInt n_aux_field_dof[n_aux_field], PetscInt m,
```

Comment:
Maybe we should use `cc_*` instead of `aux_*` in the parameters here. It might also be nice to leave some comments above this function about what it does, such as

```
// this function clones an existing DM (dm) and creates a cell-centered DM (cc_dm) with the
// given number of cell-centered fields having the indicated number of degrees of freedom
```

Also, maybe we should call `m` something like `max_field_name_len` if we're going to offer it as a parameter.